### PR TITLE
avahi - code style fixes

### PR DESCRIPTION
**Pull request by doktornotor**
avahi.xml
- Fix copyright header
- Fix indentation
- Fix permissions (none of these needs to be executable)
- Make version sane and consistent
- Hide WAN from 'Deny interfaces' selection because it's always disabled. Clarify the description note accordingly
- Move custom_php_deinstall_command to avahi.inc and make it much more thorough

avahi.inc
- Fix copyright header
- Remove useless avahi_start() / avahi_stop() functions and use start_service() / stop_service() instead
- Remove unused $g var declaration everywhere
- On install, only try to add avahi user/group if they don't exist yet
- Add avahi_deinstall() function, move the existing custom_php_deinstall_command contents here, additionally remove the avahi user/group on uninstall and also nuke /usr/local/etc/gnome.subr when removing the package
- Use full paths to binaries consistently in the rc file
- Code style fixes

Added a setting to (runtime) disable the D-BUS client API since some people are having inexplicable issues with starting avahi otherwise.

### CODE REVIEW

File: config/avahi/avahi.inc [hunk 1]
```diff
@@ -172,31 +192,29 @@ EOF;
...
+		service_stop("avahi");
+	}
 	avahi_write_config();
-
 	// Is package enabled?
-	if (($config['installedpackages']['avahi']['config'][0]['enable'])
-	    && file_exists("/usr/local/etc/rc.d/avahi-daemon.sh")) {
-		avahi_start();
+	if (($config['installedpackages']['avahi']['config'][0]['enable']) && file_exists("/usr/local/etc/rc.d/avahi-daemon.sh")) {
+		service_start("avahi");
```
Comment (phil-davis) [inline config/avahi/avahi.inc]: The function is start_service()

File: config/avahi/avahi.inc [hunk 2]
```diff
@@ -172,31 +192,29 @@ EOF;
...
 }
 
 function avahi_sync() {
-	global $g, $config;
-
-	avahi_stop();
+	global $config;
 
+	if (is_service_running("avahi")) {
+		service_stop("avahi");
```
Comment (phil-davis) [inline config/avahi/avahi.inc]: The function is start_service()

**Comment by doktornotor** [comment]
Fixed; thanks Phil :+1:

**Comment by cwagz** [comment]
consider setting enable-dbus=no in avahi.inc on line 126.  This allows the service to start at reboot in my VMs.

**Comment by doktornotor** [comment]
@cwagz: Added a checkbox to disable D-BUS

**Pull request merged**